src/client/components/editor/index.js

```diff
@@ -135,6 +135,13 @@ class Editor extends DataComponent {
       }
     });
 
+    doc.on('update', change => {
+      if( _.has( change, 'status' ) ){
+        const isDone = this.data.document.submitted() || this.data.document.published();
+        this.done( isDone );
+      }
+    });
```

Comment:
It may be a bit simpler to listen to the `status` event, but this works too.  Not sure whether the `remotestatus` event was ever hooked up.